include/podio/ObjectID.h

```diff
@@ -49,4 +50,16 @@ inline void to_json(nlohmann::json& j, const podio::ObjectID& id) {
 
 } // namespace podio
 
+template<>
+struct std::hash<podio::ObjectID>
+{
+    std::size_t operator()(const podio::ObjectID& id) const noexcept
+    {
+        auto hash_collectionID = std::hash<uint32_t>{}(id.collectionID);
```

Comment:
Technically the `collectionID` is (almost certainly) already a 32 bit hash.

Reply:
I'm not sure how that helps here. I assume, hash is still better to cover domain of size_t.